website/views/shared/tasks/task.jade

```diff
@@ -166,22 +166,56 @@ li(bindonce='list', id='task-{{::task.id}}', ng-repeat='task in obj[list.type+"s
         // if Daily, calendar
         fieldset(ng-if='::task.type=="daily"', class="option-group")
           legend.option-title=env.t('repeat')
-          ul.repeat-days(bindonce)
-            // note, does not use data-toggle="buttons-checkbox" - it would interfere with our own click binding
-            li
-              button(ng-class='{active: task.repeat.su}', type='button', ng-click='task.challenge.id || (task.repeat.su = !task.repeat.su)') {{::moment.weekdaysMin(0)}}
-            li
-              button(ng-class='{active: task.repeat.m}', type='button', ng-click='task.challenge.id || (task.repeat.m = !task.repeat.m)') {{::moment.weekdaysMin(1)}}
-            li
-              button(ng-class='{active: task.repeat.t}', type='button', ng-click='task.challenge.id || (task.repeat.t = !task.repeat.t)') {{::moment.weekdaysMin(2)}}
-            li
-              button(ng-class='{active: task.repeat.w}', type='button', ng-click='task.challenge.id || (task.repeat.w = !task.repeat.w)') {{::moment.weekdaysMin(3)}}
-            li
-              button(ng-class='{active: task.repeat.th}', type='button', ng-click='task.challenge.id || (task.repeat.th = !task.repeat.th)') {{::moment.weekdaysMin(4)}}
+
+          // Choose the frequency
+          ul.priority-multiplier
             li
-              button(ng-class='{active: task.repeat.f}', type='button', ng-click='task.challenge.id || (task.repeat.f= !task.repeat.f)') {{::moment.weekdaysMin(5)}}
+              button(ng-class='{active: task.frequency=="daily"}', type='button', ng-click='task.frequency="daily"')='Daily'
             li
-              button(ng-class='{active: task.repeat.s}', type='button', ng-click='task.challenge.id || (task.repeat.s = !task.repeat.s)') {{::moment.weekdaysMin(6)}}
+              button(ng-class='{active: task.frequency=="weekly"}', type='button', ng-click='task.frequency="weekly"')='Weekly'
+
+          // If frequency is daily
+          fieldset(ng-show='task.frequency=="daily"')
+            | Once every&nbsp;&nbsp;
+            select(ng-model='task.everyX')
+              option(ng-repeat='x in [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]') {{x}}
+            |   days<br><br>
+            | Start date:&nbsp;
+            p(class='input-group')
```

Comment:
You could create a [jade mixin](http://jade-lang.com/reference/mixins/), but I think it would make more sense to move it into a separate fieldset.